### PR TITLE
Refactor tabs to handle dependency properly

### PR DESCRIPTION
**Pull request by adam5909099**
# Task:
https://github.com/opynfinance/squeeth-monorepo/issues/292

## Description
Refactor trade tabs to optimize performance and handle dependency properly to avoid possible UI bugs like loading issues.

Fixes # (issue)
https://github.com/opynfinance/squeeth-monorepo/issues/292

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Comment by KMKoushik** [comment]
@adam5909099 Is this PR ready?

**Review by KMKoushik — APPROVED** [review]

File: packages/frontend/src/state/positions/providers.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    } else setPositionType(PositionType.NONE)
+  }, [computedSwaps.squeethAmount, setPositionType])
+
+  const value = useMemo(
+    () => ({ ...computedSwaps, squeethAmount: computedSwaps.squeethAmount.absoluteValue() }),
+    [computedSwaps],
+  )
+
+  return <ComputeSwapsContext.Provider value={value}>{children}</ComputeSwapsContext.Provider>
+}
```
Comment (EstherSeyi) [inline packages/frontend/src/state/positions/providers.tsx:110]: @adam5909099 noticed we replaced the useComputeSwaps hook with context, please can you give some context(no pun intended😂) on your thoughts behind that

**Review by block902 — APPROVED** [review]
Looks good

File: packages/frontend/src/state/positions/providers.tsx (hunk 1 above)
Comment (adam5909099) [reply packages/frontend/src/state/positions/providers.tsx:110]: useComputeSwaps is used in more than 10 places and the useMemo was called separately in each place. So I moved it into context to make sure it's called one time in global level.

**Review by EstherSeyi — APPROVED** [review]

**Pull request merged**